pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -627,43 +624,49 @@ def _getPenaltyParam(self, iw, rw):
         return xPen
 
     # fmt: off
-    def _snstop(self, ktcond, mjrprtlvl, minimize, n, nncon, nnobj, ns, itn, nmajor, nminor, nswap, condzhz, iobj, scaleobj,
-                objadd, fobj, fmerit, penparm, step, primalinf, dualinf, maxvi, maxvirel, hs, locj, indj, jcol, scales, bl, bu, fx, fcon, gcon, gobj, ycon,
-                pi, rc, rg, x, cu, iu, ru, cw, iw, rw):
+    def _snstop(self, ktcond, mjrprtlvl, minimize, n, nncon, nnobj, ns, itn, nmajor, nminor, nswap, condzhz, iobj,
+                scaleobj, objadd, fobj, fmerit, penparm, step, primalinf, dualinf, maxvi, maxvirel, hs, locj, indj,
+                jcol, scales, bl, bu, fx, fcon, gcon, gobj, ycon, pi, rc, rg, x, cu, iu, ru, cw, iw, rw):
         # fmt: on
         """
         This routine is called every major iteration in SNOPT, after solving QP but before line search
         We use it to determine the correct major iteration counting, and save some parameters in the history file.
-        If 'snSTOP function handle' is set to a function handle, then the callback is performed at the end of this function.
+        If 'snSTOP function handle' is set to a function handle, then it is called at the end of this function.
 
-        returning with iabort != 0 will terminate SNOPT immediately
+        Returns
+        -------
+        iabort : int
+            The return code expected by SNOPT. A non-zero value will terminate SNOPT immediately.
         """
         iterDict = {
             "isMajor": True,
             "nMajor": nmajor,
             "nMinor": nminor,
+            "step": step,
+            "feasibility": primalinf,
+            "optimality": dualinf,
+            "merit": fmerit,
+            "condZHZ": condzhz,
+            "penalty": penparm[2],
         }
         for saveVar in self.getOption("Save major iteration variables"):
-            if saveVar == "merit":
-                iterDict[saveVar] = fmerit
-            elif saveVar == "feasibility":
-                iterDict[saveVar] = primalinf
-            elif saveVar == "optimality":
-                iterDict[saveVar] = dualinf
-            elif saveVar == "penalty":
-                penParam = self._getPenaltyParam(iw, rw)
-                iterDict[saveVar] = penParam
+            if saveVar == "penalty_vector":
+                iterDict[saveVar] = self._getPenaltyParam(iw, rw),
```

Comment:
Maybe rename `_getPenaltyParam` to `_getPenaltyVector`? Otherwise, I think it's easy to confuse with `penparm`